core/server/master/src/main/java/alluxio/master/MasterUtils.java

```diff
@@ -57,15 +62,35 @@ public Void call() throws Exception {
   }
 
   /**
-   * @return a metastore of the configured type
+   * @return a block store of the configured type
    */
-  public static Metastore getMetaStore() {
-    Type type = Configuration.getEnum(PropertyKey.MASTER_METASTORE, Type.class);
+  public static BlockStore.Factory getBlockStoreFactory() {
```

Comment:
This means a `HeapBlockStore` is of type `BlockStore.Factory`? That sounds confusing.

Reply:
`HeapBlockStore` isn't of type `BlockStore.Factory`. This method returns a method which is a block store factory; the returned value is `args -> new HeapBlockStore(args)`